### ISSUE TITLE
How to train the whole_image_segmentation on my own datasets ?

### ISSUE THREAD

**Issue by Rudder573** [frank-xwang/UnSAM#18]
Dear Author,
I've no idea how to train the whole_image_segmentation. How should I prepare my datasets into the formal format?

**Comment by frank-xwang** [comment]
Hey, thank you for your message. You should follow our `divide_and_conquer` stage, which generates pseudo-masks for all images in your dataset with unsupervised learning methods. Please run the script `divide_and_conquer/generate_pseudo_masks.sh` — it handles the entire pseudo-mask generation process and will save all the masks produced by our unsupervised labeling pipeline. You can choose to only generate pseudo-masks for a subset of your TRAIN_DATSETS via specifying the `start-id` and `end-id`. 

Note that this script expects the following data structure:

```
TRAIN_DATASETS/
  FOLDER1/
    Image1
    .......
    Imagen
  FOLDER2/
    Image1
    .......
    Imagen
  .......
```

**Comment by Rudder573** [comment]
Thanks for answering my question so timely! Actually, I'd like to know the dataset format suitable for whole_image_segmentation training.   For instance, my dataset is currently the coco format and I also have the extra images generated by `divide_and_conquer/generate_pseudo_masks.sh`. Do I need transform it to the "sam_instance_tsv"? Another question is  FOLDER1/2 present for? Which catalogue should I put the train_datasets in? I'm quite confused about the dataset.md. Please explain it to me in detail. Your project is so intelligent. I really want to utilize it which can help me a lot.

> Hey, thank you for your message. You should follow our `divide_and_conquer` stage, which generates pseudo-masks for all images in your dataset with unsupervised learning methods. Please run the script `divide_and_conquer/generate_pseudo_masks.sh` — it handles the entire pseudo-mask generation process and will save all the masks produced by our unsupervised labeling pipeline. You can choose to only generate pseudo-masks for a subset of your TRAIN_DATSETS via specifying the `start-id` and `end-id`.
> 
> Note that this script expects the following data structure:
> 
> ```
> TRAIN_DATASETS/
>   FOLDER1/
>     Image1
>     .......
>     Imagen
>   FOLDER2/
>     Image1
>     .......
>     Imagen
>   .......
> ```

**Comment by frank-xwang** [comment]
Hey, to reduce memory usage, you can try increasing the confidence score threshold (`confidence_thresh` in `demo_whole_image.py`) and decreasing the number of proposal predictions (`NUM_OBJECT_QUERIES` in `whole_image_segmentation/configs/maskformer2_R50_bs16_50ep.yaml`). These changes can significantly lower memory consumption, though they may also reduce the total number of predicted masks (i.e. lower recall).

**Comment by Rudder573** [comment]
Thanks again for your detailed instructions! You're a really good instructor to me. 👍 But I got supernumerary predicted marks by `demo_whole_image.py` along with `pseudo_masks_output.png`. I thought that would be caused by the pseudo marks generated by `demo_dico.py`. 
And another question is I don't know how to generate pseudo instance and semantic masks on my own datasets in U2Seg project.
Please let me know at your convenience.
